Previously, cancelling the context surrounding an :func:`open_loop`
block might cause a deadlock in some cases. The ordering of operations
during loop teardown has been improved, so this shouldn't happen
anymore.
